tests/expectations/metrics/test_metrics_util.py

```diff
@@ -272,3 +274,149 @@ def test_get_unexpected_indices_for_multiple_pandas_named_indices_named_unexpect
+        batch_columns_list=test_column_names,
+    )
+    assert metric_domain_kwargs["column"] == quoted_column_name
+
+    metric_domain_kwargs = get_dbms_compatible_metric_domain_kwargs(
+        metric_domain_kwargs={"column": "travel_agents"},
+        batch_columns_list=test_column_names,
+    )
+    assert metric_domain_kwargs["column"] == quoted_column_name
+    assert metric_domain_kwargs["column"] != "TRAVEL_AGENTS"
+
+
+def test_get_dbms_compatible_metric_domain_column_pair_kwargs():
+    quoted_column_name: sqlalchemy.quoted_name = sqlalchemy.quoted_name(
+        value="travel_agents", quote=True
+    )
+    test_column_names: list[str] = [
+        "ACTORS",
+        "ARTISTS",
+        "ATHLETES",
+        "BUSINESS_PEOPLE",
+        "HEALTHCARE_WORKERS",
+        "ENGINEERS",
+        "LAWYERS",
+        "MUSICIANS",
+        "SCIENTISTS",
+        "LITERARY_PROFESSIONALS",
+        quoted_column_name,
+    ]
+
+    metric_domain_kwargs: dict
+
+    metric_domain_kwargs = get_dbms_compatible_metric_domain_kwargs(
+        metric_domain_kwargs={"column_A": "ACTORS", "column_B": quoted_column_name},
+        batch_columns_list=test_column_names,
+    )
+    assert (
+        metric_domain_kwargs["column_A"] == "ACTORS"
+        and metric_domain_kwargs["column_B"] == quoted_column_name
+    )
+
+    metric_domain_kwargs = get_dbms_compatible_metric_domain_kwargs(
+        metric_domain_kwargs={"column_A": "ACTORS", "column_B": "travel_agents"},
+        batch_columns_list=test_column_names,
+    )
+    assert (
+        metric_domain_kwargs["column_A"] == "ACTORS"
+        and metric_domain_kwargs["column_B"] == quoted_column_name
+    )
+
+
+def test_get_dbms_compatible_metric_domain_column_list_kwargs():
+    quoted_column_name_0: sqlalchemy.quoted_name = sqlalchemy.quoted_name(
+        value="travel_agents", quote=True
+    )
+    quoted_column_name_1: sqlalchemy.quoted_name = sqlalchemy.quoted_name(
+        value="FarmAnimals", quote=True
+    )
+    quoted_column_name_2: sqlalchemy.quoted_name = sqlalchemy.quoted_name(
+        value="Household_Pets", quote=True
+    )
+    test_column_names: list[str] = [
+        "ACTORS",
+        quoted_column_name_0,
+        "ARTISTS",
+        quoted_column_name_1,
+        "ATHLETES",
+        "BUSINESS_PEOPLE",
+        "HEALTHCARE_WORKERS",
+        "ENGINEERS",
+        "LAWYERS",
+        "MUSICIANS",
+        "SCIENTISTS",
+        "LITERARY_PROFESSIONALS",
+        quoted_column_name_2,
+    ]
+
+    metric_domain_kwargs: dict
+
+    metric_domain_kwargs = get_dbms_compatible_metric_domain_kwargs(
+        metric_domain_kwargs={
+            "column_list": ["ACTORS", "travel_agents", "FarmAnimals", "Household_Pets"]
+        },
+        batch_columns_list=test_column_names,
+    )
+    assert metric_domain_kwargs["column_list"] == [
+        "ACTORS",
+        quoted_column_name_0,
+        quoted_column_name_1,
+        quoted_column_name_2,
+    ]
```

Comment:
Any possible issues with sorting here? Are we guaranteed order?

Reply:
@cdkini I added sorting -- thanks.